elements/lisk-tree/src/sparse_merkle_tree/branch.ts

```diff
@@ -12,35 +12,43 @@
  * Removal or modification of this copyright notice is prohibited.
  */
 
-import { Leaf } from './leaf';
-import { branchHash } from './utils';
+import { hash } from '@liskhq/lisk-cryptography';
+import { branchDataBuffer } from './utils';
 import { NodeSide } from './constants';
 
 export class Branch {
-	private _left: Leaf;
-	private _right: Leaf;
+	private _leftHash: Buffer;
+	private _rightHash: Buffer;
 	private _hash: Buffer;
-	public constructor(left: Leaf, right: Leaf) {
-		this._left = left;
-		this._right = right;
-		this._hash = branchHash(this._left.hash, this._right.hash);
+	private _data: Buffer;
```

Comment:
maybe rename to `digest` or `digestBuffer` whichever you prefer, imo sounds better and more relatable to the implemented property.
similarly the function names in `utils` would have to be changed to `branchDigest` and so on
```suggestion
	private _digest: Buffer;
```